src/Fixer/Casing/DeclaredClassCasingFixer.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Casing;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author siad007
+ */
+final class DeclaredClassCasingFixer extends AbstractFixer
+{
+    private static $declaredClassNames;
+
+    public function __construct()
+    {
+        parent::__construct();
+
+        if (null === self::$declaredClassNames) {
+            foreach (get_declared_classes() as $class) {
```

Comment:
I'm sceptical about this line - `get_declared_classes` will not return classes that are declared, but those declared in the current script - depending how autoloader was called - which mean one day it will return some classes and another day more classes - adding some fixer to config might accidentally "discover" more classes.

When I was working on similar [fixer](https://github.com/kubawerlos/php-cs-fixer-custom-fixers/blob/v1.13.0/src/Fixer/InternalClassCasingFixer.php#L90) I have ended with only internal classes to handle.